### ISSUE TITLE
2021-11-21

### ISSUE THREAD

**Issue by ssfc** [ssfc/wangdao#662]
4314.this loss is the log loss of a (K+1)-way softmax-based classifier that tries to classify q as k+; (2021-11-1)
4325.training the encoder networks that represent the queries and keys; (2021-11-1)
433III.2. Momentum Contrast
4341.contrastive learning is building a discrete dictionary on high-dimensional continuous inputs such as images; dictionary is dynamic, key encoder evolves during training; hypothesis, a large dictionary that covers a rich set of negative samples, encoder for the dictionary keys is kept as consistent as possible; (2021-11-1)
4352.Dictionary as a queue; maintaining the dictionary as a queue of data samples; reuse the encoded keys from the immediate preceding mini-batches; queue decouples the dictionary size from the mini-batch size; (2021-11-1)
4363.The samples in the dictionary are progressively replaced; The current mini-batch is enqueued to the dictionary, and the oldest mini-batch in the queue is removed; (2021-11-1)
4374.Momentum update; intractable to update the key encoder by back-propagation; propose a momentum update to address this issue; (2021-11-1)
4386.m is a momentum coefficient; (2021-11-1)
4397.Relations to previous mechanisms;
